src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -222,21 +224,26 @@ private String acquireBearerToken() throws RestApiException {
     }
 
     @Async
-    private void executePostePayAuthorizationCall(PostePayAuthRequest postePayAuthRequest, String clientId, PaymentRequestEntity paymentRequestEntity) throws RestApiException {
+    private void executePostePayAuthorizationCall(PostePayAuthRequest postePayAuthRequest, String clientId, PaymentRequestEntity paymentRequestEntity, Boolean onboarding) throws RestApiException {
         Long idTransaction = postePayAuthRequest.getIdTransaction();
         log.info("START - execute PostePay payment authorization request for transaction " + idTransaction);
         String correlationId;
         String authorizationUrl;
         try {
             CreatePaymentRequest createPaymentRequest = createAuthorizationRequest(postePayAuthRequest, clientId);
             String bearerToken = acquireBearerToken();
-            InlineResponse200 inlineResponse200 = postePayControllerApi.apiV1PaymentCreatePost(bearerToken, createPaymentRequest);
-            if (Objects.isNull(inlineResponse200)) {
+            CreatePaymentResponse createPaymentResponse;
+            if (onboarding != null && onboarding.equals(Boolean.TRUE)) {
```

Comment:
You can use
```suggestion
            if (BooleanUtils.isTrue(isOnboarding)) {
```
this covers also the onboarding = null case